### PR TITLE
[Ready] more directional sprites for wall mounted stuff.

### PR DESCRIPTION
**Pull request by Ghommie**
:cl:
imageadd: Added directional sprites for several wall objects, such as APC and alarms.
add: Fire alarms' color display will change accordingly to the alert level now.
fix: Fixes mobs/objects with larger sprites being possibly seen over an opaque atom if standing by its corners. (e.g. an alien queen behind a wall).
/:cl:

Of course there are still a lot of wall stuff still without directionals out there, this is a first step.
The fix is necessary so we don't see wall stuff from behind the corner due to the bigger icons.

### CODE REVIEW

**Comment by Alterist** [comment]
Something something compartmentalize your prs.
remind me to change the names of the apc's to theseus instead of almayer when this gets merged and i have to replace all the apcs

**Comment by spookydonut** [comment]
split off the pure sprite change bits from the stuff refactoring reagent dispensors where possible

**Comment by DominikPanic** [comment]
Conflicts.

**Comment by Alterist** [comment]
The sprites are good. Got any screenshots of the directionals? 
please don't merge map backups

**Comment by Ghommie** [comment]
> 
> 
> The sprites are good. Got any screenshots of the directionals?
> please don't merge map backups

I'll have to fully map in their directional spawners and adjust the code furthermore before that. About the backups I'll delete them once I finish.

File: code/game/machinery/air_alarm.dm [hunk 1]
```diff
@@ -1081,26 +1090,56 @@ table tr:first-child th:first-child { border: none;}
...
+/obj/machinery/alarm/monitor/west
+	dir = WEST
+
+
+/obj/machinery/alarm/server
 	req_one_access = list(ACCESS_CIVILIAN_ENGINEERING)
+	target_temperature = 90
+
+/obj/machinery/alarm/server/first_run()
+	alarm_area = get_area(src)
```
Comment (Rohesie) [inline code/game/machinery/air_alarm.dm]: Put this after an `else`, under `if(alarm_area.master)`

File: code/game/machinery/fire_alarm.dm [hunk 2]
```diff
@@ -24,27 +24,47 @@ FIRE ALARM
 	var/wiresexposed = 0
 	var/buildstage = 2 // 2 = complete, 1 = no wires,  0 = circuit gone
 
+/obj/machinery/firealarm/New(location, direction, building)
+	..()
```
Comment (Rohesie) [inline code/game/machinery/fire_alarm.dm]:
```suggestion
	. = ..()
```

File: code/game/machinery/fire_alarm.dm [hunk 3]
```diff
@@ -232,48 +252,30 @@ FIRE ALARM
...
-		return
-	var/area/A = src.loc
-	A = A.loc
-	if (!( istype(A, /area) ))
+	if (!working)
 		return
-	A.firereset()
+	var/area/A = get_area(src)
+	if (A)
+		A.firereset()
```
Comment (Rohesie) [inline code/game/machinery/fire_alarm.dm]: `A?.firereset()`

File: code/game/machinery/fire_alarm.dm [hunk 4]
```diff
@@ -232,48 +252,30 @@ FIRE ALARM
...
+	if (!working)
 		return
-	var/area/A = src.loc
-	A = A.loc
-	if (!( istype(A, /area) ))
-		return
-	A.firealert()
+	var/area/A = get_area(src)
+	if(A)
+		A.firealert()
```
Comment (Rohesie) [inline code/game/machinery/fire_alarm.dm]: `A?.firealert()`

File: code/game/machinery/air_alarm.dm (hunk 1 above)
Comment (Ghommie) [reply code/game/machinery/air_alarm.dm]: The area is being set here. without this, it'd read null.master.
Mostly a copypasta of the parent's otherwise.

**Comment by Ghommie** [comment]
I'll probably go back at this after #788. There have been a handful of map and machinery changes within the past month.

File: code/game/objects/closets/walllocker.dm [hunk 5]
```diff
@@ -1,58 +1,272 @@
...
+	overlay_welded = "emergwelded"
+	pixel_y = 32
+	density = FALSE
+	anchored = TRUE
+	store_mobs = FALSE
+	wall_mounted = TRUE
+	large = FALSE
+	storage_capacity = 20
+
+/obj/structure/closet/secure_closet/personal/walllocker/north
```
Comment (DominikPanic) [inline code/game/objects/closets/walllocker.dm]: Don't do it like this (was discussed and deemed bad by spooky and alterist)

Comment (Ghommie) [reply code/game/objects/closets/walllocker.dm]: Then, mind enlightening me a little on would this should be done?

Comment (DominikPanic) [reply code/game/objects/closets/walllocker.dm]: Direction should be varedited. Initialize should have a dir swith that sets the pixel offsets.

Comment (Ghommie) [reply code/game/objects/closets/walllocker.dm]: okay dokay.

**Comment by Ghommie** [comment]
I'll be waiting on #875.

**Comment by Ghommie** [comment]
Whoops.

File: _maps/WIP/Sulaco_bottom.dmm [hunk 6]
```diff
@@ -460,7 +460,10 @@
 /turf/closed/wall/sulaco,
 /area/sulaco/medbay/hangar)
 "bk" = (
-/obj/structure/closet/walllocker/emerglocker/west,
+/obj/structure/closet/walllocker/emerglocker{
+	icon_state = "emerg";
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Varedited icon state.

File: _maps/WIP/Sulaco_bottom.dmm [hunk 7]
```diff
@@ -633,7 +634,10 @@
 /obj/machinery/light{
 	dir = 4
 	},
-/obj/structure/closet/walllocker/emerglocker/east,
+/obj/structure/closet/walllocker/emerglocker{
+	icon_state = "emerg";
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Same

File: _maps/WIP/Sulaco_bottom.dmm [hunk 8]
```diff
@@ -920,20 +919,18 @@
...
 	d2 = 2;
 	icon_state = "0-2";
 	pixel_y = 0
 	},
 /obj/structure/cable{
-	icon_state = "0-4";
-	d2 = 4
+	d1 = 2;
+	d2 = 4;
+	icon_state = "2-4"
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Same

File: _maps/WIP/Sulaco_bottom.dmm [hunk 9]
```diff
@@ -6079,7 +6006,10 @@
 	icon_state = "tube1";
 	dir = 8
 	},
-/obj/structure/closet/walllocker/emerglocker/west,
+/obj/structure/closet/walllocker/emerglocker{
+	icon_state = "emerg";
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Same

File: _maps/WIP/Sulaco_bottom.dmm [hunk 10]
```diff
@@ -6089,7 +6019,10 @@
 	},
 /area/shuttle/drop1/sulaco)
 "mJ" = (
-/obj/structure/closet/walllocker/emerglocker/east,
+/obj/structure/closet/walllocker/emerglocker{
+	icon_state = "emerg";
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Same

File: _maps/WIP/Sulaco_bottom.dmm [hunk 11]
```diff
@@ -6433,6 +6366,19 @@
 	},
 /turf/open/floor/engine,
 /area/sulaco/hangar/two)
+"nG" = (
+/obj/machinery/computer/secure_data/detective_computer,
+/obj/machinery/power/apc,
+/obj/structure/cable{
+	icon_state = "0-4";
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Same

File: _maps/WIP/Sulaco_bottom.dmm [hunk 12]
```diff
@@ -6433,6 +6366,19 @@
...
+	d2 = 4
+	},
+/turf/open/floor{
+	tag = "icon-carpet10-8 (WEST)";
+	icon_state = "carpet10-8";
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Same

File: _maps/WIP/Sulaco_bottom.dmm [hunk 13]
```diff
@@ -6433,6 +6366,19 @@
...
 /area/sulaco/hangar/two)
+"nG" = (
+/obj/machinery/computer/secure_data/detective_computer,
+/obj/machinery/power/apc,
+/obj/structure/cable{
+	icon_state = "0-4";
+	d2 = 4
+	},
+/turf/open/floor{
+	tag = "icon-carpet10-8 (WEST)";
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Tag

File: _maps/WIP/Sulaco_bottom.dmm [hunk 14]
```diff
@@ -6562,22 +6508,12 @@
...
-/obj/structure/cable{
-	icon_state = "0-4";
-	d2 = 4
-	},
-/turf/open/floor{
-	tag = "icon-carpet10-8 (WEST)";
-	icon_state = "carpet10-8";
-	dir = 8
+/obj/structure/closet/walllocker/emerglocker{
+	icon_state = "emerg";
```
Comment (DominikPanic) [inline _maps/WIP/Sulaco_bottom.dmm]: Varedited icon state

**Comment by Ghommie** [comment]
Cleared several varedited cases on the sulaco maps. Hopefully you are content now, that's some faults on past mappers' doings which is stressful to clear off.

File: _maps/WIP/Sulaco_bottom.dmm (hunk 8 above)
Comment (Ghommie) [reply _maps/WIP/Sulaco_bottom.dmm]: This is necessary so you can actually see how the powernet is wired while mapping.

**Comment by Rohesie** [comment]
Travis failing.
This should not be tested until a local test happens.

**Comment by Ghommie** [comment]
> 
> 
> Travis failing.
> This should not be tested until a local test happens.

Travis failing is because it detected how I asininely failed to use the map merger and not related to failure to run a local test.
It should be getting fixed by now anyway.

**Comment by Rohesie** [comment]
Please test it locally after fixing the current merge conflicts and let us know before, so we can test it again.

**Comment by Ghommie** [comment]
> 
> 
> Please test it locally after fixing the current merge conflicts and let us know before, so we can test it again.

Tested and Fixed. I've been punished by the git gods and mapmerger enough already.

**Pull request merged**